source/as-promise/types.ts

```diff
@@ -286,7 +286,7 @@ export class CancelError extends RequestError {
 	}
 }
 
-export interface CancelableRequest<T extends Response | Response['body'] = Response['body']> extends PCancelable<T>, RequestEvents<CancelableRequest<T>> {
+export interface CancelableRequest<T extends Response | Response['body'] = Response['body']> extends PCancelable<T>, Promise<T>, RequestEvents<CancelableRequest<T>> {
```

Comment:
This is actually incorrect. `PCancelable` already extends the `Promise`.

https://github.com/sindresorhus/p-cancelable/blob/4b7d014c5be44889e40014dea098ae44819b95d6/index.d.ts#L22

Reply:
Due to the interface [declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-interfaces) there is no new information for Typescript. But Webstorm's type checker can't extract information from `p-cancelable` and states `type PCancelable = any`, thus `CancelableRequest.then` returns `any`.

`CancelableRequest extends PCancelable<T>, Promise<T>` forces Webstorm to merge `type PCancelable = any` with `Promise<T>`, so `CancelableRequest.then` returns `T` now.

The reason Webstorm can't extract information from `p-cancelable` is awkward: it's not listed in the `package.json`. Compare, not listed:

<img src="https://user-images.githubusercontent.com/4586392/90688905-d58d1500-e277-11ea-9a3e-4394ce18d3a4.png" width="550">

listed:

<img src="https://user-images.githubusercontent.com/4586392/90688325-f739cc80-e276-11ea-88b7-8a456e43010a.png" width="550">

Reply:
> Due to the interface declaration merging there is no new information for Typescript

How does this relate to the actual issue? We don't use this.

> But Webstorm's type checker

See @sindresorhus's response above.

> The reason Webstorm can't extract information from p-cancelable is awkward: it's not listed in the package.json.

You're wrong:

https://github.com/sindresorhus/got/blob/eaf1e02b21e4b8c08197c6400844d4098828ecf4/package.json#L55

Reply:
* `PCancelable` extends `Promise`, so can do `CancelableRequest`
  * As typescript provides [interface declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-interfaces), there wouldn't be any new information for the compiler
  * As you write `await got(...)` in examples, you expect `CancelableRequest` to extend `Promise`
  * If `PCancelable` stops inheriting `Promise` someday, there will be a type error `CancelableRequest should implement Promise interface methods: then, catch`
  * A clear parents list improves Webstorm type inheritance

With all the above I don't really understand what may be incorrect here. A small change of code makes everyone happy and doesn't make the code worse.

> @xamgore: The reason Webstorm can't extract information from p-cancelable is awkward: it's not listed in the package.json.
> @szmarczak: You're wrong: "link to got/package.json"

If `p-cancelable` wasn't listed in `got/package.json` the project wouldn't be compilable. As you could guess, I meant `%username%/project/package.json`.